### PR TITLE
Ilnar/detail note -> detail note

### PR DESCRIPTION
**Pull request by Sunar**
includes some refactoring

### CODE REVIEW

File: app/src/main/java/com/knightua/notepadapp/ui/fragments/main/MainFragmentPresenter.kt [hunk 1]
```diff
@@ -62,90 +64,90 @@ class MainFragmentPresenter : BasePresenter<MainFragmentView>(),
     }
 
     override fun detach() {
-        super.detach()
         unregisterReceivers()
```
Comment (Sunar) [inline app/src/main/java/com/knightua/notepadapp/ui/fragments/main/MainFragmentPresenter.kt:67]: В unregisterReceivers удаляются подписки у view, но в super.detach он уже стал null... Memory leak

File: app/src/main/java/com/knightua/notepadapp/ui/fragments/main/MainFragmentPresenter.kt [hunk 2]
```diff
@@ -178,32 +183,39 @@ class MainFragmentPresenter : BasePresenter<MainFragmentView>(),
...
-        }
-    }
-
-    private fun handleData(notes: List<Note>) {
-        Timber.i(notes.toString())
-        if (isInternetRelay.value!!) {
-            NotepadApp.injector.getNoteRepository().insertAllInDatabase(notes)
-            dataRelay.accept(Pair(dataRelay.value?.first?.plus(1)!!, notes))
-        }
+    private fun loadData() {
```
Comment (Sunar) [inline app/src/main/java/com/knightua/notepadapp/ui/fragments/main/MainFragmentPresenter.kt:186]: отдельный метод для загрузки из API

File: app/src/main/java/com/knightua/notepadapp/ui/fragments/main/MainFragmentPresenter.kt [hunk 3]
```diff
@@ -62,90 +64,90 @@ class MainFragmentPresenter : BasePresenter<MainFragmentView>(),
...
+                dataRelay.accept(it)
+            }
+            .flatMap {
+                NotepadApp.injector.getNoteRepository().getAllFromApi().toFlowable()
+                    .doOnError {
+                        if (dataRelay.hasValue() && dataRelay.value?.isEmpty() == true) {
+                            stateRelay.accept(STATE_NO_CONNECTION)
+                        } else {
+                            stateRelay.accept(STATE_DATA)
+                            getView()?.showSnackbarError(R.string.error_no_connection)
```
Comment (Sunar) [inline app/src/main/java/com/knightua/notepadapp/ui/fragments/main/MainFragmentPresenter.kt:133]: getView() внутри dataCompositeDisposable не рекомендуется, потому что dataCompositeDisposable живет дольше view, и getView() может возвращать null. Но как я уже говорил, мы просто пытаемся показать снэкбар, если экран не открыт, то не он покажется, но ничего в этом страшного
